extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js

```diff
@@ -1161,3 +1294,48 @@ function getFirstInstanceValue_(instances, extractFn) {
   }
   return null;
 }
+
+/**
+ * Maps a sentinel value to an object consisting of the impl to which that
+ * sentinel value belongs and the corresponding message handler for that impl.
+ * @type{!Object<string, !Object>} */
+const fluidListeners = {};
+let listenerForFluidRegistered = false;
+
+/**
+ * @param {!AmpAdNetworkDoubleclickImpl} instance
+ */
+function registerListenerForFluid(instance) {
```

Comment:
Let's make this a class level function so that you don't need to pass instance.  Just make fluidListeners a "static" field such that when empty you know that you need to add the postMessage listeners (can then remove listenerForFluidRegistered).

if (!Object.keys(fluidListeners).length) this.win.addEventListener...

Reply:
Done.